marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/Spent.hs

```diff
@@ -133,6 +133,50 @@ mkSpentIndexer path = do
     [spentInsert]
     [Core.SQLRollbackPlan "spent" "slotNo" C.chainPointToSlotNo]
 
+catchupConfigEventHook :: Trace IO Text -> FilePath -> Core.CatchupEvent -> IO ()
+catchupConfigEventHook stdoutTrace dbPath Core.Synced = do
+  SQL.withConnection dbPath $ \c -> do
+    let slotNoIndexName = "spent_slotNo"
+        createSlotNoIndexStatement =
+          "CREATE INDEX IF NOT EXISTS "
+            <> fromString slotNoIndexName
+            <> " ON spent (slotNo)"
+    createIndexTable c slotNoIndexName createSlotNoIndexStatement
+
+    let txInIndexName = "spent_txIn"
+        createTxInIndexStatement =
+          "CREATE INDEX IF NOT EXISTS "
+            <> fromString txInIndexName
+            <> " ON spent (txId, txIx)"
+    createIndexTable c txInIndexName createTxInIndexStatement
+
+    let spentAtTxIdIndexName = "spent_spentAtTxId"
+        createSpentAtIndexStatement =
+          "CREATE INDEX IF NOT EXISTS "
+            <> fromString spentAtTxIdIndexName
+            <> " ON spent (spentAtTxId)"
+    createIndexTable c spentAtTxIdIndexName createSpentAtIndexStatement
+  where
+    createIndexTable :: SQL.Connection -> String -> SQL.Query -> IO ()
+    createIndexTable c indexName createIndexStatement = do
+      (doesIndexExist :: Bool) <-
+        liftIO $
+          fmap ((> (0 :: Int)) . sum . concat) $
+            SQL.query_
+              c
+              ( [sql|SELECT COUNT(*)
+                  FROM sqlite_master
+                  WHERE type='index' AND name='|]
+                  <> fromString indexName
+                  <> "'"
+              )
+      unless doesIndexExist $ do
+        logInfo stdoutTrace $
+          "Creating SQL index table '" <> Text.pack indexName <> "' for Spent indexer..."
+        liftIO $ SQL.execute_ c createIndexStatement
+        logInfo stdoutTrace $
+          "Finished creating SQL index table '" <> Text.pack indexName <> "' for Spent indexer..."
```

Comment:
This is slightly duplicated across some indexers, but I don't see the ideal place to put it.

Reply:
Add a helper function to either `Core.Marconi.Indexer.SQLite` or `Core.Marconi.Transformer.WithCatchup`?

Reply:
I guess I can put it in `Core.Marconi.Indexer.SQLite`, but feels a bit out-of-place.

Reply:
In a submodule of one of these two?

Reply:
I put it in `Marconi.Core.Transformer.WithCatchup.SQLite`.